engine/cluster/src/main/java/org/opennms/alec/engine/cluster/GraphManager.java

```diff
@@ -168,6 +177,10 @@ private synchronized void addOrUpdateInventory(Collection<InventoryObject> inven
 
         // Update the set of disconnected vertices
         trackDisconnectedVertices(verticesToVerify);
+
+        if(haveWeChangedGraph.get()) {
```

Comment:
Could also call `didGraphChange.set(true)` in this block and remove the other statements beside the `haveWeChangedGraph.set(true)` calls.

Reply:
agree, it looks a bit confusing that we hold twice essentially same state. It would be nice if we could simplify that.
didGraphChange represents the overall state of the graph and is reset regularly by calling _getDidGraphChangeAndReset_ via the Driver Thread
_haveWeChangedGraph_ reresents a marker within the method if the method itself has changed the graph and if so it will call the graphChangedListeners at the end of the method

@j-white let's discuss in our call if it is ok to merge the two variables.

Reply:
fixed